pkg/tablecodec/tablecodec.go

```diff
@@ -975,20 +975,33 @@ func decodeHandleInIndexKey(keySuffix []byte) (kv.Handle, error) {
 	return kv.NewCommonHandle(keySuffix)
 }
 
-func decodeHandleInIndexValue(value []byte) (kv.Handle, error) {
+func decodeHandleInIndexValue(value []byte) (handle kv.Handle, err error) {
+	var seg IndexValueSegments
 	if getIndexVersion(value) == 1 {
-		seg := SplitIndexValueForClusteredIndexVersion1(value)
-		return kv.NewCommonHandle(seg.CommonHandle)
+		seg = SplitIndexValueForClusteredIndexVersion1(value)
+	} else {
+		if len(value) <= MaxOldEncodeValueLen {
```

Comment:
Maybe `getIndexVersion(value) == 0` is more readable, Or you can set a temporary variable for "getIndexVersion". Mainly to unify with line980.

Reply:
Comments updated.